pkg/certificate/providers/mrcclient.go

```diff
@@ -0,0 +1,31 @@
+package providers
+
+import (
+	"fmt"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/pkg/errors"
+)
+
+type MRCClientImpl struct {
```

Comment:
Avoid using the name impl in go. Instead, describe what the type is, it doesn't need to know about the interface it is implementing.

What's backing this type? Is it an informer? A message.Broker? A compat layer?

MRCInformerClient, MRCBrokerClient, MRCCompatClient, etc.

Reply:
Removed the entire data structure